lib/get-editor-config.js

```diff
@@ -0,0 +1,230 @@
+// code parts copypasted from https://github.com/editorconfig/editorconfig-core-js
+
+'use strict';
+const micromatch = require('micromatch');
+const path = require('path');
+const fs = require('fs');
+
+class EditorConfigResolver {
+  constructor() {
+    this.EDITOR_CONFIG_CACHE = null;
+    this.CONFIG_RESOLUTIONS_CACHE = new Map();
+    /**
+     * define the possible values:
+     * section: [section]
+     * param: key=value
+     * comment: ;this is a comment
+     */
+    this.regex = {
+      section: /^\s*\[(([^#;]|\\#|\\;)+)\]\s*([#;].*)?$/,
+      param: /^\s*([\w.\-_]+)\s*[=:]\s*(.*?)\s*([#;].*)?$/,
+      comment: /^\s*[#;].*$/,
+    };
+
+    this.knownProps = {
+      end_of_line: true,
+      indent_style: true,
+      indent_size: true,
+      insert_final_newline: true,
+      trim_trailing_whitespace: true,
+      charset: true,
+    };
+  }
+  opts(filepath, options) {
+    const resolvedFilePath = path.resolve(filepath);
+    return [resolvedFilePath, this.processOptions(options, resolvedFilePath)];
+  }
+  processOptions(options = {}, filepath) {
+    return {
+      config: options.config || '.editorconfig',
+      root: path.resolve(options.root || path.parse(filepath).root),
+    };
+  }
+  parseString(data) {
+    let sectionBody = {};
+    let sectionName = null;
+    const value = [[sectionName, sectionBody]];
+    const lines = data.split(/\r\n|\r|\n/);
+    lines.forEach(line => {
+      let match;
+      if (this.regex.comment.test(line)) {
+        return;
+      }
+      if (this.regex.param.test(line)) {
+        match = line.match(this.regex.param);
+        sectionBody[match[1]] = match[2];
+      } else if (this.regex.section.test(line)) {
+        match = line.match(this.regex.section);
+        sectionName = match[1];
+        sectionBody = {};
+        value.push([sectionName, sectionBody]);
+      }
+    });
+    return value;
+  }
```

Comment:
![image](https://user-images.githubusercontent.com/1360552/68314835-e7764080-00c7-11ea-904e-fed98f120026.png)

parse in editorconfig its = find all editorconfigs, compile and find related config for passed file name

Reply:
@rwjblue `Lets use ini package for parsing the .editorconfig file` - this parsing logic originally from `editorconfig` package, don't think it will change

Reply:
don't want to have alot of side deps for this functionality